mathics/builtin/forms/output.py

```diff
@@ -188,12 +189,16 @@ def eval_makeboxes(self, expr, f, evaluation):
         # MakeBoxes is formatting an expression inside a DisplayForm.
         # Hopefully, this is temporal and it is not going to be
         # needed after the Format/MakeBoxes refactor.
-        previous_df, evaluation.in_display_form = evaluation.in_display_form, True
+
+        old_value_in_display_form = SYMBOL_FORMATBOXES.evaluate(evaluation)
+        evaluation.definitions.set_ownvalue(SYMBOL_FORMATBOXES.name, SymbolTrue)
```

Comment:
Notice that here we can not use the higher level way
```
Expression(SymbolSet, SYMBOL_NO_FORMATBOXES, SymbolTrue).evaluate(evaluation)
```
because the attributes.

Reply:
The drawbacks of this approach are that the access to the attribute is slower, and the code to set the context is larger and weirder than just setting an attribute.